software-engineering-policies/Containers/ContainerPolicy.md

```diff
@@ -1,55 +1,23 @@
-* Define resources/limits/requests for the CPU/memory etc limits without these you cannot auto scale.
-* Utilise IAC aka Helm/Kustomize to deploy/configure clusters/environments.
-* Health checks akak readiness and liviness probe
-* Ensure HTTPS is enforced for environments above Dev and Http -> Https redirect is enabled
-* Ensure container logs and consumed by something
-* Block access in above Dev to the physical machines running the cluster/ AKS employ vnet/firewall/RBAC rules etc
-* Container Dependency Vulnerability Scanning?
-* Use minimal dependencies, such as “Distroless” container base images:  <https://github.com/GoogleContainerTools/distroless>.
-* Avoid unnecessary privileges and Do not run as root
-* Have a process for updating dependencies.
-
-### Should do
+## Motivation
 
-* Zero trust, Microservices/containers should communicate only with who they need to locked down by policy (ingress/egress)
-* Access Policies, Network security Groups should be applied
-* Namespace all containers
-* Monitoring solution - ELK (.Net) EFK (Java) stacks, Graphana, Prome...(learn to spell)
-* Utilise automated deployments on version increments within the container registry out to the nominated environment (eg dev)
+Containers are in use throughout Data Science both in development, test and production environments. This policy defines the use and deployment for current and future usage.
 
-### Could do
+## Policy 
 
-* Utilise DAPR for sidecar third party component management (queues, DB's etc)
-* Utilise native built in support within visual studio for Docker
-* Utilise Docker Compose to stub other dependencies that container utilises
-
-## Best practice guides
-
-<https://sysdig.com/blog/dockerfile-best-practices/>
-
-Some notes from a previous principal dev on Docker/containers
-
-1. Skills - we should look to keep stacks quite similar amongst the different teams within Software Engineering (without stifling innovation), to facilitate rotation but also the transfer of services from team to support team. I would not want a delivery to be using containers and the destination support team to have no skills in them. This could arguably be covered as part of the handover and up front support design of a solution, but may differ from team to team. Do we need formal training in containers across SE? Or probably at the very least guidance on the use of, to prevent teams achieving the same thing in 100 different ways?
+* Containers are a potential option for architects to utilise (a tool in the toolbox).
+* Consideration should be made around why we want to utilise containers.
+* Consideration should be made around who will inherit the environments at the outset. i.e. is there expertise in how manage the hosting environment?
+* Utilisation of Containers in development and test environments is acceptable without APF review.
+* Use of a containerised product should be considered on a case-by-case basis and should form part of the APF approval process
```

Comment:
Previously we said: * Use of docker into live will need to be approved by The Head of Software Engineering in * unction with the Principal and Lead Software Engineers, until agreement is reached via APF forum.

This appears to have change the policy for containers in production/live to be just APF approval without input from Head of Software Engineering/Principals. Is that intended?

Reply:
No, I think this was the text I picked up. In this case, I am going to swap it to what you suggested